### PR TITLE
lib: bin: lwm2m_carrier: update to version 0.21.0 

### PR DESCRIPTION
**Pull request by nordic-hani**
Updated library to version 0.21.0

### CODE REVIEW

**Comment by jtguggedal** [comment]
@nordic-hani There's a couple more places where disabling the LTE fallback option can be removed:

```
nrf/applications/asset_tracker/lwm2m_carrier_overlay.conf:
  23  # Disable NB-IoT fallback
  24: CONFIG_LTE_NETWORK_USE_FALLBACK=n

nrf/samples/nrf9160/mqtt_simple/overlay-carrier.conf:
  23  # Disable NB-IoT fallback
  24: CONFIG_LTE_NETWORK_USE_FALLBACK=n
  ```

**Review by umapraseeda — APPROVED** [review]

**Review by lucasvdp — APPROVED** [review]

File: doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst [hunk 1]
```diff
@@ -41,30 +41,45 @@ The :ref:`lwm2m_carrier` sample project configuration (:file:`nrf/samples/nrf916
...
-
 * :kconfig:`CONFIG_LWM2M_CARRIER_CERTIFICATION_MODE`:
 
   * This configuration specifies if the LwM2M carrier library will connect to the carrier's certification servers or production servers.
   * Expected to be set during self-testing part of the certification process.
-  * This configuration is ignored if :kconfig:`CONFIG_LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_URI` is set.
+  * This configuration is ignored if :kconfig:`CONFIG_LWM2M_CARRIER_USE_CUSTOM_SERVER_URI` is set.
 
-* :kconfig:`CONFIG_LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_URI`:
+* :kconfig:`CONFIG_LWM2M_CARRIER_USE_CUSTOM_SERVER_URI`:
```
Comment (tokangas) [inline doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst:50]: Should these two be CONFIG_LWM2M_CARRIER_USE_CUSTOM_URI instead?

File: doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst [hunk 2]
```diff
@@ -41,30 +41,45 @@ The :ref:`lwm2m_carrier` sample project configuration (:file:`nrf/samples/nrf916
...
 
-  * This configuration, together with :kconfig:`CONFIG_LWM2M_CARRIER_CUSTOM_BOOTSTRAP_URI` lets the LwM2M carrier library connect to a bootstrap server other than the normal carrier server, thereby enabling the generic mode.
+  * This configuration, together with :kconfig:`CONFIG_LWM2M_CARRIER_USE_CUSTOM_URI` lets the LwM2M carrier library connect to a custom server other than the normal carrier server, thereby enabling the generic mode.
```
Comment (tokangas) [inline doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst:52]: And this one perhaps CONFIG_LWM2M_CARRIER_CUSTOM_URI?

File: applications/asset_tracker_v2/overlay-carrier.conf [hunk 3]
```diff
@@ -15,9 +15,6 @@ CONFIG_NRF_MODEM_LIB_SYS_INIT=n
 # The library is only certified for LTE-m
```
Comment (tokangas) [inline applications/asset_tracker_v2/overlay-carrier.conf:15]: I see that other NB-IoT "restrictions" have been removed, so just wondering whether this comment is still valid?

**Review by 57300 — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -535,6 +535,28 @@ Libraries for Zigbee
...
+
+* Updated:
+
+  * Updated :ref:`nrf_modem` to version 1.4.0.
+    See the :ref:`nrfxlib:nrf_modem_changelog` for detailed information.
+  * nrf_errno values have been aligned with the errno values of newlibc C library.
+  * The :ref:`Modem API <nrf_modem_api>` (:file:`nrf_modem.h`) has been updated to return negative errno values on error.
+  * The :ref:`Full Modem DFU API <nrf_modem_full_dfu_api>` (:file:`nrf_modem_full_dfu.h`) has been updated to return negative errno values on error.
+  * The :ref:`GNSS API <nrf_modem_gnss_api>` (:file:`nrf_modem_gnss.h`) has been updated to return negative errno values on error.
+
```
Comment (jtguggedal) [inline doc/nrf/releases/release-notes-changelog.rst:554]: @lemrey It looks like `nrf_modem_gnss_init` is remove in this release, should that be mentioned here?

File: doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst (hunk 1 above)
Comment (nordic-hani) [reply doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst]: Thanks

File: applications/asset_tracker_v2/overlay-carrier.conf (hunk 3 above)
Comment (nordic-hani) [reply applications/asset_tracker_v2/overlay-carrier.conf]: Removed the limitation (and comment).

File: doc/nrf/releases/release-notes-changelog.rst (hunk 4 above)
Comment (lemrey) [reply doc/nrf/releases/release-notes-changelog.rst:554]: That's just a summary of the most relevant changes. The complete changelog is in `nrfxlib`, and it mentions that.
I don't think we need to add it here.

Comment (jtguggedal) [reply doc/nrf/releases/release-notes-changelog.rst:555]: I see, then it's all good

**Review by jtguggedal — APPROVED** [review]

File: doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst (hunk 2 above)
Comment (nordic-hani) [reply doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst:52]: Thanks

**Review by eibe85 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by lemrey — APPROVED** [review]

File: lib/at_sms_cert/Kconfig [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	  AT+CSCA
+	  AT+CSCS
+	  AT+CMGD
+	  AT+CMSS
+	  AT+CMGW
+
+if AT_SMS_CERT
+
+config AT_SMS_CERT_SYS_INIT
+	bool "Initialize during SYS_INIT"
```
Comment (tejlmand) [inline lib/at_sms_cert/Kconfig:22]: curious, what would be the alternative ?

If user disables this then how will it be initialized, and can that be done in a safe manner.

Basically what i'm asking is whether it makes sense that this setting is directly user-configurable.

**Review by tejlmand — COMMENTED** [review]
looks good.

A minor question.

**Comment by lemrey** [comment]
Updated SHAs

File: lib/at_sms_cert/Kconfig (hunk 5 above)
Comment (lemrey) [reply lib/at_sms_cert/Kconfig:22]: Good point, in general the user either wants to compile this feature or not. However it should be possible to toggle the feature at runtime when it's compiled, we'll need a header for that.

Comment (tejlmand) [reply lib/at_sms_cert/Kconfig:22]: > However it should be possible to toggle the feature at runtime when it's compiled, we'll need a header for that.

So until that header is ready, maybe we should remove this setting, or as minimal make it prompt-less.
When the header is reader, then we can allow the user to either initialize this through sysinit (this setting) or runtime.

**Pull request merged**